2009-10-14  Michael Tautschnig  <mt@debian.org>

	* setup-storage, setup-storage/Exec.pm: Hide udevsettle call
Index: trunk/bin/setup-storage
===================================================================
--- trunk.orig/bin/setup-storage
+++ trunk/bin/setup-storage	
@@ -192,11 +192,14 @@
 $FAI::udev_settle = "udevadm settle --timeout=10" if (&FAI::in_path("udevadm"));
 $FAI::udev_settle = "udevsettle --timeout=10" if (&FAI::in_path("udevsettle"));
 defined($FAI::udev_settle) or die "Failed to find determine a proper way to tell udev to settle; is udev installed?";
-&FAI::execute_with_udevsettle($FAI::commands{$_}{cmd}) foreach (&numsort(keys %FAI::commands));
+foreach (&numsort(keys %FAI::commands)) {
+  `$FAI::udev_settle`;
+  &FAI::execute_command($FAI::commands{$_}{cmd});
+}
 
 # generate the proposed fstab contents
 # wait for udev to set up all devices
-&FAI::execute_with_udevsettle("true");
+`$FAI::udev_settle`;
 my @fstab = &FAI::generate_fstab(\%FAI::configs);
 
 # print fstab
Index: trunk/lib/setup-storage/Exec.pm
===================================================================
--- trunk.orig/lib/setup-storage/Exec.pm
+++ trunk/lib/setup-storage/Exec.pm	
@@ -237,19 +237,6 @@
   }
   return "";
 }
-################################################################################
-#
-# @brief Execute a command as in execute_command, but prefix it with udevsettle
-#
-# @return the identifier of the error
-#
-################################################################################
-sub execute_with_udevsettle {
-  my ($command, $stdout, $stderr) = @_;
-  defined ($FAI::udev_settle) or &FAI::internal_error("udev settle command not defined");
-  return &execute_command("$FAI::udev_settle && $command", $stdout,
-    $stderr);
-}
 
 ################################################################################
 #
